std/json.d

```diff
@@ -1206,7 +1206,7 @@ if (isInputRange!T && !isInfinite!T && isSomeChar!(ElementEncodingType!T))
                 }
 
                 value.type_tag = JSONType.string;
-                value.store.str = str;
+                () @trusted { value.store.str = str; } ();
```

Comment:
I think it would be simpler to replace both lines with `assign(str)`, which already has a `() @trusted { store.str = t; }();`.

Reply:
done